File: ET3ALEM_SERVER/BusinessEntities/Models/QuizQuestion.cs
using System.ComponentModel.DataAnnotations;

namespace BusinessEntities.Models
{
    public class QuizQuestion
    {
        public int Id { get; set; }
        public int Sequence { get; set; }

        [Required] public int QuestionId { get; set; }

        [Required] public int QuizId { get; set; }

        public Question Question { get; set; }
        public int Grade { get; set; }
    }
}

File: ET3ALEM_CLIENT_WEB/src/app/quiz/Model/quizQuestion.ts
import { Transform, Type } from 'class-transformer';
import { Question } from 'src/app/question/Models/question';
import { QuestionTypeResolver } from 'src/app/question/shared/question-type-resolver';

export class QuizQuestion {
    Id: number;
    @Transform((value, obj, type) => QuestionTypeResolver.getSpecificQuestion(value))
    Question: Question;
    Grade: number;
    QuestionId: number;
    Sequence: number;

    constructor(question: Question=null, grade = 0, Id = 0, Sequence = 0) {
        this.Question = question;
        this.Grade = grade;
        this.Id = Id;
        this.QuestionId = question?.Id;
        this.Sequence = Sequence;
    }
}